### ISSUE TITLE
Query docs show return types as `<query_name as QueryConfig<TyCtxt<'tcx>>>::Stored`

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#77459]
e.g. https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.hir_crate

This makes the docs really hard to read: you have to click on `hir_crate`, open `impl<'tcx> QueryConfig<TyCtxt<'tcx>> for hir_crate`, and know that `Value` is what actually matters, not `Stored`. This is super confusing and not documented anywhere. We should find a way to just put `&'tcx Crate<'tcx>` directly in the return type instead.

**Comment by jyn514** [comment]
`TyCtxtAt` has the same problem: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/query/struct.TyCtxtAt.html#method.hir_crate

**Comment by jyn514** [comment]
I'm not sure what's going on with `Stored` here:
[`eval_to_const_value_raw`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.eval_to_const_value_raw) returns [`EvalToConstValueResult`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/interpret/error/type.EvalToConstValueResult.html) (not a borrow), while
[`generics_of`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.generics_of) returns [`&ty::Generics`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Generics.html) (a borrow). However in the query file both of them are listed as returning an owned value.

**Comment by jyn514** [comment]
I think the difference is that `generics_of` uses [`ArenaCache`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_query_system/query/caches/struct.ArenaCache.html#impl-QueryCache), which returns a borrow? ugh

**Comment by jyn514** [comment]
Maybe the fix should be for rustdoc to turn `<X as Y>::Assoc` into the actual type that describes? That would make the standard library docs easier to read too.

**Comment by jyn514** [comment]
Re-opening this until I fix https://github.com/rust-lang/rust/issues/79459 properly.

**Comment by jyn514** [comment]
This was fixed for rustc by https://github.com/rust-lang/rust/pull/79525. I guess we can reuse this as the tracking issue for enabling `normalize-docs` globally? I'm not planning to work on it myself but I'd love to remove the feature gate.

**Comment by jyn514** [comment]
Closing this in favor of https://github.com/rust-lang/rust/issues/82692, which has proper links and such so it's clear what this is blocked on.
